### PR TITLE
Update the Docs landing page for the CodeQL docs site

### PR DESCRIPTION
**Pull request by felicitymay**
These updates reflect the migration of documentation for CodeQL CLI and CodeQL for VS Code to the GitHub docs site.

In addition, the following changes are included:

- Heading levels on the landing page adjusted for accessibility (no skipped levels)
- Minor changes to the footer on the landing page to avoid redirects in links to content that's moved
- New article on Running CodeQL queries written and added to the landing page
- Template updated to match changes to the landing page

### CODE REVIEW

**Comment by felicitymay** [comment]
To aid with reviewing, here are a couple of screenshots:

New docs landing page and revised drop-down menu:

Screenshot showing the content of the landing page and the drop-down menu. Demonstrates the new heading levels.

Revised footer for the docs landing page:

Screenshot showing the appearance of the footer with the new heading level.

New article:
Screenshot showing the preview of the new article on using CodeQL tools to run queries.

**Review by coadaflorin — APPROVED** [review]
This looks good to me. I was mostly reviewing the images you shared (Thank you for doing that!).

File: docs/codeql/writing-codeql-queries/running-codeql-queries.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+Running CodeQL queries
+======================
+
+There are several options available for running one or more CodeQL queries on a codebase. The best option depends on what your aims are.
+
+Work through a CodeQL tutorial
+------------------------------
+
+If you're working through a CodeQL tutorial, the CodeQL extension for Visual Studio Code makes it easy to run the queries in the tutorial. Unless you want to run the query on a specific code base, it's easiest to run queries on one of the many CodeQL databases that are available on GitHub. To get started, see "`Installing CodeQL for Visual Studio Code <https://docs.github.com/en/code-security/codeql-for-vs-code/getting-started-with-codeql-for-vs-code/installing-codeql-for-vs-code>`__".
```
Comment (jc-clark) [inline docs/codeql/writing-codeql-queries/running-codeql-queries.rst:13]: Slight suggestion to make it a bit more beginner friendly. 

```suggestion
If you're working through a CodeQL tutorial, the CodeQL extension for Visual Studio Code allows you to run the queries in the tutorial. Unless you want to run the query on a specific code base, it's easiest to run queries on one of the many CodeQL databases that are available on GitHub. To get started, see "`Installing CodeQL for Visual Studio Code <https://docs.github.com/en/code-security/codeql-for-vs-code/getting-started-with-codeql-for-vs-code/installing-codeql-for-vs-code>`__".
```

File: docs/codeql/writing-codeql-queries/running-codeql-queries.rst [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Develop a new CodeQL query
+--------------------------
+
+If you're developing a new query, the CodeQL extension for Visual Studio Code makes it easy to run a query and compare the results with previous runs as you refine the query. The extension also provides autocomplete suggestions, syntax highlighting, and other features that make it easier to write and debug queries. To get started, see "`Installing CodeQL for Visual Studio Code <https://docs.github.com/en/code-security/codeql-for-vs-code/getting-started-with-codeql-for-vs-code/installing-codeql-for-vs-code>`__".
```
Comment (jc-clark) [inline docs/codeql/writing-codeql-queries/running-codeql-queries.rst:18]: Similar to above. 

```suggestion
If you're developing a new query, the CodeQL extension for Visual Studio Code allows you to run a query and compare the results with previous runs as you refine the query. The extension also provides autocomplete suggestions, syntax highlighting, and other features that make it easier to write and debug queries. To get started, see "`Installing CodeQL for Visual Studio Code <https://docs.github.com/en/code-security/codeql-for-vs-code/getting-started-with-codeql-for-vs-code/installing-codeql-for-vs-code>`__".
```

**Review by jc-clark — APPROVED** [review]
This looks great from a docs content perspective! Thank you for the screenshots, those were very helpful ❤️

**Comment by felicitymay** [comment]
@coadaflorin or @jc-clark - can I ask for another approval? Committing Joe's review suggestions dismissed your reviews.

I _think_ that the update to the docs landing page will be published when we release CodeQL CLI 2.19.1.

**Review by coadaflorin — APPROVED** [review]

**Pull request merged**